File: Day02_Activity/app/src/main/java/day01/nguyenhongphat0/day02_activity/InputActivity.java
package day01.nguyenhongphat0.day02_activity;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class InputActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_input);
    }

    public void clickToDone(View view) {
        EditText editUsername = findViewById(R.id.editUsername);
//        Intent intent = new Intent(InputActivity.this, MainActivity.class);
        Intent intent = this.getIntent();
        intent.putExtra("username", editUsername.getText().toString());
//        startActivity(intent);
        this.setResult(RESULT_OK, intent);
        finish();
    }
}

File: kotlin/Day02/app/src/main/java/day02/nguyenhongphat0/day02/InputActivity.kt
package day02.nguyenhongphat0.day02

import android.app.Activity
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v7.app.AppCompatActivity;
import android.widget.EditText

import kotlinx.android.synthetic.main.activity_input.*

class InputActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_input)
        setSupportActionBar(toolbar)

        fab.setOnClickListener { view ->
            val username = findViewById<EditText>(R.id.usernameInput).text.toString()
            intent.putExtra("username", username)
            setResult(Activity.RESULT_OK, intent)
            finish()
        }
    }

}
